Signal/src/ViewControllers/HomeView/HomeViewController.m

```diff
@@ -221,9 +221,10 @@ - (void)loadView
     UILabel *emptyBoxLabel = [UILabel new];
     self.emptyBoxLabel = emptyBoxLabel;
     [self.view addSubview:emptyBoxLabel];
-    [emptyBoxLabel autoPinWidthToSuperview];
-    [emptyBoxLabel autoPinToTopLayoutGuideOfViewController:self withInset:0];
-    [emptyBoxLabel autoPinToBottomLayoutGuideOfViewController:self withInset:0];
+    [emptyBoxLabel setNumberOfLines:0]; //  Let the label use as many lines as needed. It will very rarely be more than 2.
+    [emptyBoxLabel autoPinLeadingToSuperviewMargin];
+    [emptyBoxLabel autoPinTrailingToSuperviewMargin];
+    [[[emptyBoxLabel centerYAnchor] constraintEqualToAnchor:[[self view] centerYAnchor]] setActive:YES];
```

Comment:
We use PureLayout (or our own PureLayout inspired methods) for making AL constraints.

So this should be something like:

```
[emptyBoxLabel autoAlignAxis:ALAxisHorizontal toSameAxisOfView:self.view];
```

Or maybe just:
```
[emptyBoxLabel autoVCenterInSuperview];
```

Reply:
I got the impression that PureLayout is being slowly phased out from Signal since it doesn't seem to be in active development and the newer Signal code isn't using it, plus the improvements in auto layout API through the years have gotten it to the point where auto layout wrappers aren't really that much of a benefit for most common use cases.

I thought about coming up with a patch to remove usage of the PureLayout library to begin with, but that would be a lot of code churn and require a firm decision of the best way to go forward. I'd advocate for just using the modern UIKit APIs for all but the more complex cases, but the wrappers you're building are ok if a bit redundant for a few too many things for my taste. If neither proved a problem I could go ahead with that, on a separate patch of course. FWIW I've been accused to knowing my auto layouts 👍🏽

I also didn't know if you'd want a utility method for this single use case, but it's an easy enough change to make. I'll get that in place in a second.

Reply:
> I got the impression that PureLayout is being slowly phased out from Signal since it doesn't seem to be in active development and the newer Signal code isn't using it

Nope, we're still using PureLayout.

The `autoVCenterInSuperview` method is already defined, can you use the existing definition and remove your newly added method?